mem/kv_mem.go

```diff
@@ -645,6 +648,27 @@ func (kv *memKV) Unlock(kvp *kvdb.KVPair) error {
 	return err
 }
 
+func (kv *memKV) EnumerateWithSelect(
+	prefix string,
+	enumerateSelect kvdb.EnumerateSelect,
+	copySelect kvdb.CopySelect,
+) ([]interface{}, error) {
+	var kvi []interface{}
+	prefix = kv.domain + prefix
+	for k, v := range kv.m {
```

Comment:
need to protect access to kv.m from concurrent read-write access. can you double check all accesses while you are at it ?

Reply:
will go test -race help?

Reply:
Yes --race helped.
However it also showed a lot of race conditions in our test code.
In the latest commit I have handled all the race conditions reported for non test related code.